### DISCUSSION TITLE
Weigh lidar returns by distance

### DISCUSSION THREAD

**Discussion by hidmic** [Ekumen-OS/beluga#339] [Ideas]
Jotting a down a suggestion that got brought up internally. The farther out a lidar hit is reported, the larger the effect of measurement, estimation, mapping errors in beam and likelihood model performance. We could downplay these effects by scaling measurement log likelihood with distance $e^{-d_i}$ (can this be understood as scaling the amount of information that the model can provide :thinking:?).

CC @glpuga @nahueespinosa to check if my recollections are accurate.

**Comment by nahueespinosa** [comment]
I believe the suggestion revolves around amplifying the importance of measurements taken at greater distances from the robot. This adjustment aims to address the fact that nearby obstacles often generate numerous lidar returns, which can overshadow the significance of the fewer, but potentially crucial, measurements obtained from farther away. Credit to @glpuga, and please correct me if I got the idea wrong.

At some point, there was a suggestion to incorporate the arc length between two consecutive lidar returns as a factor when aggregating the individual ray weights. This approach would not only consider the distance of each measurement but also account for the spatial relationship between consecutive returns, but I'm probably missing details now.

Sorry I didn't wrote this down before (or explain it better).

**Reply by hidmic** [reply]
> I believe the suggestion revolves around amplifying the importance of measurements taken at greater distances from the robot.

Uh, that's counter-intuitive. Downplaying nearby measurements would amplify the effects of both measurement and mapping errors, which worsen with distance.

> This adjustment aims to address the fact that nearby obstacles often generate numerous lidar returns, which can overshadow the significance of the fewer, but potentially crucial, measurements

That is a fair point, and partially due to practical deviations in both beam and likelihood model definitions, that I think speaks to a deeper issue. Not all beams bear useful information and the informative subset changes over time. If we could prune non-informative measurements, the system would be more precise (and perhaps but not necessarily more accurate) and faster.

I wonder if there's research about this :thinking:

**Comment by hidmic** [comment]
FYI @glpuga
